<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GameObjectScriptComponentWidgetQt</class>
 <widget class="QWidget" name="GameObjectScriptComponentWidgetQt">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>312</width>
    <height>405</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>312</width>
    <height>405</height>
   </size>
  </property>
  <property name="acceptDrops">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>Game Object Script Components</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QMainWindow, QWidget, QDialog, QMenuBar::item
{
	background-color: rgb(50, 50, 50);
	color: rgb(255, 255, 255);
}

QHeaderView::section
{
	background-color: qlineargradient(spread:reflect, x1:0, y1:0, x2:0, y2:0.5, stop:0 rgba(50, 50, 50, 255), stop:1 rgba(75, 75, 75, 255));
	padding-left: 4px;
	border: 1px solid rgb(110, 110, 110);
}

QMenuBar::item:selected, QMenu::item:selected
{
    background-color: rgb(120, 120, 120);
	border: 1px solid rgb(200, 200, 200);
}

QPushButton:hover
{
    background-color: rgb(75, 75, 75);
}

QPushButton:pressed
{
    background-color: rgb(100, 100, 100);
}

QLineEdit
{
	border-width: 2px;
	border-style: inset;
	border-color: rgb(60, 60, 60);
	border-radius: 5px;
	padding: 1px;
}

QLineEdit:focus
{
	background-color: rgb(75, 75, 75);
}

QDockWidget::title
{
	background-color: rgb(65, 65, 65);
	padding: 5px;
}

QTabBar::tab 
{
	background-color: qlineargradient(spread:pad, x1:1, y1:0.5, x2:0, y2:0.5, stop:0 rgba(60, 60, 60, 255), stop:0.5 rgba(80, 80, 80, 255), stop:1 rgba(60, 60, 60, 255));
	padding: 5px;
	borde-width: 3px; 
	border-color: rgb(110, 110, 110);
	border-style: inset;
	border-radius: 8px;
	font-size: 11px;
}

QTabBar::tab:selected 
{
	background-color: qlineargradient(spread:pad, x1:1, y1:0.5, x2:0, y2:0.5, stop:0 rgba(60, 60, 60, 255), stop:0.5 rgba(120, 120, 120, 255), stop:1 rgba(60, 60, 60, 255));
	border-style: outset;
    border-bottom-color: rgb(60, 60, 60);
}

QTreeWidget
{
	alternate-background-color: rgb(60, 60, 60);
}

QTreeWidget::branch:has-children
{
	image: url(:/XEngine_Editor/Resources/Images/branch-closegrey.png) 0;
}

QTreeWidget::branch:has-children:open
{
	image: url(:/XEngine_Editor/Resources/Images/branch-opengrey.png) 0;
}

QTreeWidget::branch:has-children:open:selected
{
	image: url(:/XEngine_Editor/Resources/Images/branch-openblue.png) 0;
}

QTreeWidget::branch:has-children:!open:selected
{
	image: url(:/XEngine_Editor/Resources/Images/branch-closeblue.png) 0;
}
</string>
  </property>
  <widget class="QTreeWidget" name="m_GameObjectScriptTree">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>40</y>
     <width>291</width>
     <height>81</height>
    </rect>
   </property>
   <property name="autoScrollMargin">
    <number>0</number>
   </property>
   <property name="alternatingRowColors">
    <bool>false</bool>
   </property>
   <property name="indentation">
    <number>0</number>
   </property>
   <property name="rootIsDecorated">
    <bool>true</bool>
   </property>
   <property name="itemsExpandable">
    <bool>false</bool>
   </property>
   <property name="sortingEnabled">
    <bool>false</bool>
   </property>
   <column>
    <property name="text">
     <string>Scripts</string>
    </property>
   </column>
  </widget>
  <widget class="QPushButton" name="m_AddGOScript">
   <property name="geometry">
    <rect>
     <x>130</x>
     <y>10</y>
     <width>81</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Add Script</string>
   </property>
  </widget>
  <widget class="QPushButton" name="m_RemoveGOScript">
   <property name="geometry">
    <rect>
     <x>220</x>
     <y>10</y>
     <width>81</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Remove Script</string>
   </property>
  </widget>
  <widget class="GameObjectScriptPropertyWidget" name="m_GameObjectScriptProperties">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>192</y>
     <width>291</width>
     <height>201</height>
    </rect>
   </property>
   <property name="autoScrollMargin">
    <number>0</number>
   </property>
   <property name="alternatingRowColors">
    <bool>false</bool>
   </property>
   <property name="indentation">
    <number>0</number>
   </property>
   <property name="rootIsDecorated">
    <bool>true</bool>
   </property>
   <property name="itemsExpandable">
    <bool>false</bool>
   </property>
   <property name="sortingEnabled">
    <bool>false</bool>
   </property>
   <column>
    <property name="text">
     <string notr="true">1</string>
    </property>
   </column>
  </widget>
  <widget class="QLineEdit" name="m_ScriptAssetTextBox">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>131</y>
     <width>191</width>
     <height>20</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Script Asset</string>
   </property>
   <property name="readOnly">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QLabel" name="m_ScriptAssetLabel">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>132</y>
     <width>61</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Script Asset:</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="m_InstanceNameTextBox">
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>162</y>
     <width>151</width>
     <height>20</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Vertex Shader</string>
   </property>
   <property name="readOnly">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QLabel" name="m_InstanceNameLabel">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>163</y>
     <width>81</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Instance Name:</string>
   </property>
  </widget>
  <widget class="QPushButton" name="m_SearchScriptAsset">
   <property name="geometry">
    <rect>
     <x>280</x>
     <y>130</y>
     <width>21</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>...</string>
   </property>
  </widget>
  <widget class="QPushButton" name="m_CreateRemoveInstance">
   <property name="geometry">
    <rect>
     <x>250</x>
     <y>160</y>
     <width>51</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Create</string>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GameObjectScriptPropertyWidget</class>
   <extends>QTreeWidget</extends>
   <header>Widgets\Custom\GameObjectScriptPropertyWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
